src/preamble.js

```diff
@@ -752,15 +752,17 @@ function instrumentWasmExportsWithAbort(exports) {
   for (var name in exports) {
     (function(name) {
       var original = exports[name];
-      
+
       // Wrap all functions, copy the other symbols.
-      if (typeof original === 'function') {
+      // Don't wrap the asan helper functions since they are used during
+      // checkStackCookie and would causes our tests to fail.
```

Comment:
```suggestion
      // checkStackCookie which would causes our tests to fail.
```

Reply:
Why would they fail? Does asan call them after exit?

Reply:
The problem is test_abort_on_exceptions:

```
      addOnPostRun(function() {                                                                         
        try {                                                                                           
          // Crash the program                                                                          
          _crash();                                                                                     
        }                                                                                               
        catch(e) {                                                                                      
          // Catch the abort                                                                            
          out(true);                                                                                    
        }                                                                                               
                                                                                                        
        out("again");                                                                                   
                                                                                                        
        try {                                                                                           
          // Try executing some function again                                                          
          _crash();                                                                                     
        }                                                                                               
        catch(e) {                                                                                      
          // Make sure it failed with the expected exception                                            
          out(e === "program has already aborted!");                                                    
        }                                                                                               
      });    
```

It tries to trigger the exception which makes the module unusable, but handle the exception and exit cleanly.

The problem is that checkStackCookie always runs at the end of `run()` which is normally harmless, but in asan mode it calls back into the module.

The alternative I guess would be to have checkStackCookie check if the module has crashed in just NOP if it has?

Reply:
Ok that worked!  Maybe better this way?

Reply:
Yes, I think that's more correct.